src/button-group.ts

```diff
@@ -115,17 +115,43 @@ export default class GlideCoreButtonGroup extends LitElement {
     `;
   }
 
-  #orientation: 'horizontal' | 'vertical' = 'horizontal';
+  #defaultSlotElementRef = createRef<HTMLSlotElement>();
 
-  #slotElementRef = createRef<HTMLSlotElement>();
+  #orientation: 'horizontal' | 'vertical' = 'horizontal';
 
   #variant?: 'icon-only';
 
   get #buttonElements() {
     return [...this.querySelectorAll('glide-core-button-group-button')];
   }
 
-  #onSlotChange() {
+  #onDefaultSlotChange() {
+    const areMultipleButtonsSelected =
+      this.#buttonElements.filter(({ selected }) => selected).length > 1;
+
+    if (areMultipleButtonsSelected) {
+      // With form controls where only one control should be selected, like Radio Group or
+      // single-select Dropdown, we allow consumers to set multiple options as selected:
+      // both to match native and because we're able to avoid negative outcomes, at the form
+      // level, by ensuring the `value` of those components only includes one value.
+      //
+      // Button Group is different because its state doesn't end at a form. Its state is always
+      // represented somewhere else in the application. And, if the data or logic used to
+      // produce Button Group's state results in multiple selected buttons, then the same data
+      // or logic may also produce multiple selected tables or panes, for example.
+      //
+      // We could easily account for multiple selected buttons, like we do in other components,
+      // by leaving every button selected and only showing the last one as selected visually.
+      // But there's no guarantee that the logic responsible for showing one table or pane
+      // instead of another is the same as Button Group. Maybe the first selected table or pane
+      // is shown. Then there would be a mismatch between Button Group and another part of the
+      // application.
+      //
+      // Long story short, that's why we throw. Because multiple selected buttons indicates bad
+      // downstream data or logic that's likely to have consequences outside of Button Group.
+      throw new Error('Only one selected Button Group Button is allowed.');
```

Comment:
> Button Group now throws in development

I'm not seeing where this `throw` would only happen in development... Wouldn't this hit in prod too?

Reply:
Derp. That's right. I'd gotten to used to the behavior of `assertSlot()` and `@required`. I'll adjust the changeset.